### ISSUE TITLE
Standardization of ready command

### ISSUE THREAD

**Issue by Leo-Nogismus** [IIIM-IS/AERA#155]
Since we have a tcp-connection with an external environment simulator now, I would suggest that we agree on a standard for the ejection of the ready command from AERA. This way the tcp_io_device can check, if environment and seed are the correct combination.

Right now when the ready command is defined we use the following code:
```
start:(pgm |[] |[] |[] []
   (cmd ready [<name> <entitiy>] 1)
1) |[]
; Instantiate the start program to run once.
istart:(ipgm start |[] RUN_ONCE sampling_period VOLATILE NOTIFY 1) [[SYNC_ONCE now 0 forever stdin nil 1]]
```
Which is then used to assign the entity object to the variables used in later update of the environment (both in test_mem.cpp and in tcp_io_device.cpp). However, if multiple entities are used we must manage their ejection. Additionally it would be nice to know, when using the tcp-io-device, that the environment and the seed actually match. This removes unnecessary problems if the wrong replicode file is used in the settings.xml file by accident.
For this I would suggest to a) include an identifier for the seed in the ready command, and b) use identifiers for the entities, so that we can asign them to the correct objects in the io-device code.

So for example (please suggest better solutions, if you have any):
```
start:(pgm |[] |[] |[] []
   (cmd ready [<seed_identifier> <entitiy1-name> <entitiy1> <entitiy2-name> <entitiy2> ... <entitiyN-name> <entitiyN>] 1)
1) |[]
; Instantiate the start program to run once.
istart:(ipgm start |[] RUN_ONCE sampling_period VOLATILE NOTIFY 1) [[SYNC_ONCE now 0 forever stdin nil 1]]
```

**Comment by jefft0** [comment]
Hi @Leo-Nogismus . With your recent work integrating Webots, I wonder if you have further thoughts on this issue. If your recent code addresses some of the issues that you raise above, can you mention them here?

**Comment by Leo-Nogismus** [comment]
No, the work on Webots did not encounter this problem again. This has something to do with "who is responsible for the entity creation". As we have it right now in the Webots project, both sides need to ensure that the same entities exist and can be used. If not, we will get errors (at the latest during runtime). I honestly forgot about this issue... I would still love to have a way to ensure that both sides know about the same stuff before starting. Currently, Webots sends the setup message with all entities to the AERA system and there, they are mapped to the entities in the seed. But if you would load in the wrong .replicode file, it would throw errors (or at least a warning) that the received entities do not exist in the seed (afaik).

I think that if we want to approach this whole problem, we should first think about a more "AERAesc" definition of what an entity is. As it is, if I understood this correctly, we cannot create entities in AERA during runtime, so we might want to have a look at how we want to represent objects like the sphere or the cube inside AERA (go away from an entity and move to sensor data). But this is a whole different thing.
From there we could move forward, use entities to represent sensors of the system and then move forward to ensure that the seed fits the sensors.

Sorry for the rumbling, just my ideas regarding this matter.
